pkg/registry/registry.go

```diff
@@ -30,123 +30,116 @@ const (
-	return &ianaRegistry{
-		registry: reg,
-	}
-}
-
-func NewAntreaRegistry() *antreaRegistry {
-	reg := make(map[string]entities.InfoElement)
-	globalReg[AntreaEnterpriseID] = make(map[uint16]entities.InfoElement)
-	return &antreaRegistry{
-		registry: reg,
-	}
-}
-
-func (reg *ianaRegistry) registerInfoElement(ie entities.InfoElement) error {
-	if _, exist := reg.registry[ie.Name]; exist {
-		return fmt.Errorf("IANA Registry: Information element %s has already been registered", ie.Name)
-	}
-	reg.registry[ie.Name] = ie
-	globalReg[IANAEnterpriseID][ie.ElementId] = ie
-	reverseIE, err := reg.GetReverseInfoElement(ie.Name)
-	if err == nil { // the information element has reverse information element
-		globalReg[ReverseEnterpriseID][ie.ElementId] = *reverseIE
-	}
-	return nil
-}
-
-func (reg *ianaRegistry) GetInfoElement(name string) (*entities.InfoElement, error) {
+func GetIANAInfoElement(name string) (*entities.InfoElement, error) {
 	var exist bool
 	var ie entities.InfoElement
-	if ie, exist = reg.registry[name]; !exist {
+	if ie, exist = ianaRegistry[name]; !exist {
 		err := fmt.Errorf("IANA Registry: There is no information element with name %s", name)
 		return &ie, err
 	}
 	return &ie, nil
 }
 
-func (reg *ianaRegistry) GetReverseInfoElement(name string) (*entities.InfoElement, error) {
+func GetIANAReverseIE(name string) (*entities.InfoElement, error) {
 	var exist bool
 	var ie entities.InfoElement
-	if ie, exist = reg.registry[name]; !exist {
+	if ie, exist = ianaRegistry[name]; !exist {
 		err := fmt.Errorf("IANA Registry: There is no information element with name %s", name)
 		return &ie, err
 	}
 	if !isReversible(ie.Name) {
 		err := fmt.Errorf("IANA Registry: The information element %s is not reverse element", name)
 		return &ie, err
 	}
-	reverseName := "reverse" + strings.Title(ie.Name)
+	reverseName := "reverse_" + strings.Title(ie.Name)
 	return entities.NewInfoElement(reverseName, ie.ElementId, ie.DataType, ReverseEnterpriseID, ie.Len), nil
 }
 
-func (reg *antreaRegistry) registerInfoElement(ie entities.InfoElement) error {
-	if _, exist := reg.registry[ie.Name]; exist {
-		return fmt.Errorf("Antrea Registry: Information element %s has already been registered", ie.Name)
-	}
-	reg.registry[ie.Name] = ie
-	globalReg[AntreaEnterpriseID][ie.ElementId] = ie
-	return nil
-}
-
-func (reg *antreaRegistry) GetInfoElement(name string) (*entities.InfoElement, error) {
+func GetAntreaInfoElement(name string) (*entities.InfoElement, error) {
 	var exist bool
 	var ie entities.InfoElement
-	if ie, exist = reg.registry[name]; !exist {
+	if ie, exist = antreaRegistry[name]; !exist {
 		err := fmt.Errorf("Antrea Registry: There is no information element with name %s", name)
 		return &ie, err
 	}
 	return &ie, nil
 }
 
-func (reg *antreaRegistry) GetReverseInfoElement(name string) (*entities.InfoElement, error) {
+func GetAntreaReverseIE(name string) (*entities.InfoElement, error) {
```

Comment:
This is not correct--I think we missed this before. We cannot use ReverseEnterpriseID for Antrea registry, that is only applicable for IANA registry. 
https://tools.ietf.org/html/rfc5103#section-6.2

We should define reverse elements as part of Antrea registry itself. I don't think we have any use case for Antrea reverse elements. Let me know if you think otherwise.

Reply:
Got it. I don't think we are using Antrea reverse elements. Thanks!